src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -549,7 +549,7 @@ class column_store
         }
         auto bo = *base_offset_iter;
         auto ix = base_offset_iter.index();
-        auto hint_it = _hints.find(bo);
+        auto hint_it = _hints.lower_bound(bo);
```

Comment:
seems like maybe we should have a unit test for the case that caught this issue

Reply:
It's difficult to model since it's just an optimization for faster lookup. I can see if there is a significant time discrepancy, but it would require some code to work as a baseline.

probably it could be easier to expose some internals to the test, maybe via a friend declaration